src/Workspaces/Core/Portable/Workspace/Solution/ProjectState.cs

```diff
@@ -821,20 +833,29 @@ public ProjectState RemoveDocuments(ImmutableArray<DocumentId> documentIds)
 
     public ProjectState RemoveAdditionalDocuments(ImmutableArray<DocumentId> documentIds)
     {
+        if (documentIds.IsEmpty)
+            return this;
+
         return With(
             projectInfo: ProjectInfo.WithVersion(Version.GetNewerVersion()),
             additionalDocumentStates: AdditionalDocumentStates.RemoveRange(documentIds));
     }
 
     public ProjectState RemoveAnalyzerConfigDocuments(ImmutableArray<DocumentId> documentIds)
     {
+        if (documentIds.IsEmpty)
+            return this;
+
         var newAnalyzerConfigDocumentStates = AnalyzerConfigDocumentStates.RemoveRange(documentIds);
 
         return CreateNewStateForChangedAnalyzerConfigDocuments(newAnalyzerConfigDocumentStates);
     }
 
     public ProjectState RemoveAllDocuments()
     {
+        if (DocumentStates.IsEmpty && AdditionalDocumentStates.IsEmpty && AnalyzerConfigDocumentStates.IsEmpty)
+            return this;
+
         // We create a new CachingAnalyzerConfigSet for the new snapshot to avoid holding onto cached information
         // for removed documents.
         return With(
```

Comment:
so this means we dont' get a new ProjectState with a new version.  I think that's likely ok.